libvast/vast/detail/flat_map.hpp

```diff
@@ -19,7 +19,7 @@ namespace vast::detail {
 
 template <class Key, class T, class Compare>
 struct flat_map_policy {
-  static bool pair_compare(const std::pair<const Key, T>& x, const Key& y) {
+  static bool pair_compare(const std::pair<Key, T>& x, const Key& y) {
```

Comment:
Isn't the key `const` in the `value_type` or a map?

Reply:
For `std::map` yes, for `detail::vector_map` no:

    // vector_map.hpp
    using value_type = std::pair<Key, T>;

The previous version would actually trigger the copy constructor of `std::pair`:

    /usr/include/c++/10/ext/new_allocator.h:150:4: error: use of deleted function ‘constexpr std::pair<_T1, _T2>::pair(const std::pair<_T1, _T2>&) [with _T1 = const vast::qualified_record_field; _T2 = std::unique_ptr<vast::synopsis>]’

Reply:
I think it would be a good idea to change the definition of `flat_map` to make a `vector_map` with `const Key` as key type, but I'll leave that to a future refactoring.